lib/rubocop/cop/rspec/rails/minitest_assertions.rb

```diff
@@ -229,6 +252,45 @@ def assertion
               'be_empty'
             end
           end
+
+          MSG = 'Use `%<prefer>s`.'
+
+          ASSERTION_MATCHERS = constants(false).filter_map do |c|
+            const = const_get(c)
+
+            const if const.is_a?(Class) && const.superclass == BasicAssertion
+          end
+
+          RESTRICT_ON_SEND = ASSERTION_MATCHERS.flat_map { |m| m::MATCHERS }
+
+          ASSERTION_MATCHERS.each do |m|
+            name = m.name.split('::').last
+
+            def_node_matcher "minitest_#{name}".to_sym, m::NODE_MATCHER_PATTERN
+          end
+
+          def on_send(node)
+            ASSERTION_MATCHERS.each do |m|
+              name = m.name.split('::').last
+
+              public_send("minitest_#{name}".to_sym, node) do |*args|
+                on_assertion(node, m.match(*args))
+              end
+            end
+          end
+
+          def on_assertion(node, assertion)
+            return if assertion.nil?
```

Comment:
Can it be nil? ‘match` delegates to new?

Reply:
it delegates in _most_ cases, but this check allows more advanced bail outs that cannot be expressed with node patterns alone - i.e. for checking `_predicate` we bail if the method does not end with `?` and for `in_delta` we bail if the `delta` is missing but the `failure_message` is present (which indicates an `assert_in_delta x, y, "message"` type call).

Reply:
Aha, got it. Still, I feel it’s a bit off.
Would you still feel it’s good if we checked if it’s nil or not before passing it to on_assertion a few rows above?

Reply:
Yup I agree - the only reason I had it there earlier is I was hoping I might be able to pass in the method as the block, but seems that can't be done with class methods.

Done in https://github.com/rubocop/rubocop-rspec/pull/1790/commits/4fe5c518b3a2039027cddd5923faf600b7c64636